frontend/src/components/header/Header.tsx

```diff
@@ -12,10 +12,10 @@ function Header() {
   const resetUser = useAdminStore((state) => state.resetUser);
   const resetJobStore = useJobStore((state) => state.resetJobStore);
   const navigate = useNavigate();
-  const displayName = currentUser.displayName.split(' ');
+  const displayName = currentUser?.displayName?.split(' ');
 
-  const formattedFirstName = getFormattedName(displayName[0]);
-  const formattedLastName = getFormattedName(displayName[1]);
+  const formattedFirstName = getFormattedName(displayName && displayName[0]);
```

Comment:
now it will return undefined if the value is not valid, is this ok? Another approach could be to use `Nullish Coalescing Operator` this will check whether the value is null. and you can use it as `displayName[0] ?? ""` (this returns an empty string if the value is invalid.

Reply:
oh I see, and I forgot that we are handling invalid values within `getFormattedName` 🙏